### PR TITLE
[DEV-1015], [DEV-1033], [DEV-1048] Poolpair liquidity check adjustments and other improvements

### PR DESCRIPTION
**Pull request by TonySemikin**
#### Pre-Release

- [x] Check migrations
  - No database related infos (sqldb-xxx)
  - Impact on GS (new/removed columns)
- [x] Check for linter errors (in PR)
- [x] Test basic user operations
  - Login/logout
  - Registration
  - Update user data
  - Create/delete payment routes

#### Post-Release

- Test basic user operations
- Monitor application insights log

### CODE REVIEW

File: src/subdomains/core/buy-crypto/process/entities/buy-crypto.entity.ts [hunk 1]
```diff
@@ -257,4 +281,21 @@ export class BuyCrypto extends IEntity {
             : null,
         };
   }
+
+  //*** HELPER METHODS ***//
+
+  private resetTransaction(): this {
+    this.outputReferenceAmount = null;
+    this.outputReferenceAsset = null;
```
[the remark this answers is not in the document]
Comment (TonySemikin) [reply src/subdomains/core/buy-crypto/process/entities/buy-crypto.entity.ts:289]: nope, its set before in previous steps, and now we need to save buy-crypto without batch, so everything that was generated by batching process have to be reset

File: src/subdomains/core/buy-crypto/process/entities/buy-crypto.entity.ts [hunk 2]
```diff
@@ -206,10 +215,25 @@ export class BuyCrypto extends IEntity {
...
+  waitingForLowerFee(): this {
+    this.resetTransaction();
+    this.status = BuyCryptoStatus.WAITING_FOR_LOWER_FEE;
+
+    return this;
+  }
+
+  recordFee(fee: BuyCryptoFee): this {
+    this.fee = fee;
+    this.status = BuyCryptoStatus.IN_PROGRESS;
```
[the remark this answers is not in the document]
Comment (TonySemikin) [reply src/subdomains/core/buy-crypto/process/entities/buy-crypto.entity.ts:227]: this is final step of batching preparation, so it kinda makes sense to set InProgress here. Or I could put this in first step where we define asset pair, in both cases I would not move this to separate method or so. recordFeeAndStartBuyCrypto also a bit misleading. TBH I don't see an issue in this case

[the remark this answers is not in the document]
Comment (TonySemikin) [reply src/subdomains/core/buy-crypto/process/entities/buy-crypto.entity.ts:227]: maybe even opposite, now we know that buy crypto is in progress when fee is recorded just looking at entity

[the remark this answers is not in the document]
Comment (davidleomay) [reply src/subdomains/core/buy-crypto/process/entities/buy-crypto.entity.ts:227]: Ok, makes sense 👍

File: src/subdomains/core/buy-crypto/process/entities/buy-crypto.entity.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (davidleomay) [reply src/subdomains/core/buy-crypto/process/entities/buy-crypto.entity.ts:289]: Ok, cool

**Review by davidleomay — APPROVED** [review]

**Pull request merged**